app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -66,6 +66,16 @@ def workflow_helper
     @workflow_helper ||= MiqProvisionOrchWorkflow.new({:src_vm_id => [id]}, User.current_user, :skip_dialog_load => true, :initial_pass => true)
   end
 
+  def target_name_valid?(name, ems_folder_id = nil)
+    # TODO: need a way to tell the ovf template is for VM template (where target is a VM) or vApp template (where target is a resource pool with VMs)
+    folder = EmsFolder.find_by(:id => ems_folder_id)
+    if folder
+      folder.vms.none? { |vm| vm.name == name }
```

Comment:
This seems expensive...can we do it with a query?

```ruby
folder.vms.where(:name => name).exists?
```

Reply:
`folder.vms` returns an array. @Fryguy

Reply:
@lfu from this [line](https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_folder.rb#L18), I believe that `folder.vms` returns an ActiveRecord relationship, not an Array.

Reply:
@chessbyte folders use `Relationships` so yeah it is an array unfortunately
```
>> EmsFolder.find_by(:name => "vm").vms.class.name
=> "Array"
```

Reply:
I can't say if it is by design @Fryguy might know, but it is how `Relationship` have always worked afaik

Reply:
This is the nature of the relationships table, though there are some things to deal with some of the performance issues